### ISSUE TITLE
1.2.154.0 fails vkvia and crashes due to "invalid storage mode"

### ISSUE THREAD

**Issue by miyehn** [KhronosGroup/MoltenVK#1093]
Hello,

I downloaded VulkanSDK 1.2.154.0 and followed the installation instructions but here's the command line output when running `vkvia`:
```
VIA_INFO:    SDK Found! - Will attempt to run tests
VIA_INFO:       Attempting to run ./vkcube in /Users/miyehn/Documents/github/vulkansdk/macOS/../Applications/vkcube.app/Contents/MacOS
VIA_INFO:           Command-line: ./vkcube --c 100 --suppress_popups
validateMTLStorageMode:237: failed assertion `invalid storageMode (0). Must be one of MTLStorageModeManaged(1) MTLStorageModePrivate(2)'
VIA_INFO:       Attempting to run ./cube in /Users/miyehn/Documents/github/vulkansdk/macOS/../Applications/vkcube.app/Contents/MacOS
VIA_INFO:           Command-line: ./cube --c 100 --suppress_popups
VIA_WARNING: ./cube not found.  Skipping.
VIA_ERROR:   Unknown Test failure occurred.
```
And [here](https://github.com/KhronosGroup/MoltenVK/files/5338657/vkcube_2020-10-07-105841_RAINDU-MB0.txt)'s a crash report.

My device is Macbook Pro 16', with OS version 10.15.5, and graphics cards AMD Radeon Pro 5500M 8 GB and Intel UHD Graphics 630 1536 MB.

I uninstalled it and installed 1.2.148.1 instead and it works fine, so it's probably something new. If someone's going to look into this let me know if there's anything else I can provide :)

**Comment by cdavis5e** [comment]
Apple told me `Shared` textures would work in 10.15.5...

Does it work if you upgrade to 10.15.6?

**Comment by miyehn** [comment]
Ah I upgraded to 10.15.7 and it worked. Thanks!

**Comment by cdavis5e** [comment]
So I need to tweak the condition that enables `Shared` linear textures, then, as I thought. Thanks for verifying that.

**Comment by billhollings** [comment]
@cdavis5e 

>So I need to tweak the condition that enables Shared linear textures, then, as I thought. Thanks for verifying that.

You'll take care of this?

>Apple told me Shared textures would work in 10.15.5...

Where is this documented? The standard Apple docs [here](https://developer.apple.com/documentation/metal/mtlbuffer/1613852-newtexturewithdescriptor?language=occ) and [here](https://developer.apple.com/documentation/metal/mtlstoragemode/mtlstoragemodeshared?language=occ) give no indication (even as Beta) that coherent textures are supported on macOS.

I'm thinking about how best to handle issue #175, both in terms of documenting the limitation (the latest complaint) and an eventual solution to coherent textures on macOS.

**Comment by cdavis5e** [comment]
> @cdavis5e
> 
> > So I need to tweak the condition that enables Shared linear textures, then, as I thought. Thanks for verifying that.
> 
> You'll take care of this? Should we move it to 10.15.6 or 10.15.7?

Yes, I have a change to bump this to 10.15.6, for now. If we discover it doesn't work on 10.15.6, then we can bump it to 10.15.7.

> 
> > Apple told me Shared textures would work...
> 
> Where is this documented...even for 10.15.7 or 11.0? The standard Apple docs [here](https://developer.apple.com/documentation/metal/mtlbuffer/1613852-newtexturewithdescriptor?language=occ) and [here](https://developer.apple.com/documentation/metal/mtlstoragemode/mtlstoragemodeshared?language=occ) give no indication (even as Beta) that coherent textures are supported on macOS.
> 
> I'm thinking about how best to handle issue #175, both in terms of documenting the limitation (the latest complaint) and an eventual solution to coherent textures on macOS.

It isn't. I filed a feedback, and they told me that it was fixed in 10.15.5.

**Comment by billhollings** [comment]
>I have a change to bump this to 10.15.6, for now

Thanks.

>It isn't. I filed a feedback, and they told me that it was fixed in 10.15.5.

Meh. Okay. It is what it is from Apple.

I haven't tested this myself. But I'd like to address this in documentation as pointed out in issue #175. What was "fixed"? Are we able to say that permanently-open `vkMapMemory()` calls on coherent (Shared) texture memory are stable and usable as coherent now on iOS and macOS 10.15.7)+?

We should probably also add a `hostCoherentTextureMemory` element to the Portability Initiative feature list too.

After you confirm status, I'll take care of both of these changes (MVK docs & portability).

**Comment by cdavis5e** [comment]
> What was "fixed"? Are we able to say that permanently-open `vkMapMemory()` calls on coherent (Shared) texture memory are stable and usable as coherent now on iOS and macOS 10.15.7)+?

Yes.
